### PR TITLE
ENH: Use highspy in linprog.. again

### PR DESCRIPTION
**Pull request by HaoZeke**
This is a newer variant of #19255. It is meant to work with https://github.com/ERGO-Code/HiGHS/pull/1749, or the SciPy local https://github.com/scipy/HiGHS/pull/66

Do not merge until:

- [ ] The right scipy/highs commit is used (post https://github.com/ERGO-Code/HiGHS/pull/1749)

### CODE REVIEW

**Comment by mdhaber** [comment]
Could you please include the tests of gh-20589 in this PR? If they pass, we can also mark this as closing the issues linked there.

**Comment by HaoZeke** [comment]
> Could you please include the tests of [gh-20589](https://github.com/scipy/scipy/pull/20589) in this PR? If they pass, we can also mark this as closing the issues linked there.

It seems that `test_large_numbers_gh20116` and `test_bug_20336` still don't work.

**Comment by mdhaber** [comment]
Hmm ok. Are you able to try those problems out with pure HiGHS code (no SciPy wrapper) and (assuming the problem persists) submit an upstream bug report? Perhaps the tests could still be incorporated here, but with an `xfail` and link to the upstream issues?

**Comment by rgommers** [comment]
I merged the needed `scipy/highs` PR. It'd be useful to get this in, and not blocking on new regression tests (what works now is nice, the rest can be kept for follow-up in gh-20589), since this is already a huge PR that we had to revert once before.

File: .gitmodules [hunk 1]
```diff
@@ -26,3 +22,7 @@
 [submodule "scipy/_lib/cobyqa"]
 	path = scipy/_lib/cobyqa
 	url = https://github.com/cobyqa/cobyqa.git
+[submodule "subprojects/highs"]
+	path = subprojects/highs
+	url = https://github.com/HaoZeke/highs
```
Comment (rgommers) [inline .gitmodules:27]: @HaoZeke can you update this?

**Comment by rgommers** [comment]
A number of CI jobs are now failing with:
```
 /opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/scipy/optimize/_highs/_highs_wrapper.py:4: in <module>
    import scipy.optimize._highs.highspy as _h
/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/scipy/optimize/_highs/highspy/__init__.py:3: in <module>
    from highspy._core import \
E   ModuleNotFoundError: No module named 'highspy'
```

Not sure why they're not all failing - the absolute `import highspy` isn't going to work.

**Comment by HaoZeke** [comment]
Closing in favor of https://github.com/scipy/scipy/pull/21565

**Pull request closed without merging**